### PR TITLE
Do not generate default repetition ids if use_repetition_ids=False

### LINKED ISSUE #5418: Don't construct CircuitOperation huge repetition id lists when use_repetition_ids=False
Do not run this code unless you like going out of memory.

```
import cirq
op = cirq.CircuitOperation(
    cirq.FrozenCircuit(),
    repetitions=1_000_000_000_000,
    use_repetition_ids=False)
```

# Expected behavior

Because `use_repetition_ids` is set to `False`, no list of repetition ids is ever created and the code finishes in under a second.

# Actual behavior

Internally, the code attempts to create a list of one trillion repetition ids. This list will never be used.

---

For context, I know that Mike Newman has used stim to run circuits with hundreds of millions of rounds. If he ever attempted to use stimcirq to convert that circuit into a cirq circuit... `*KABOOM*`.

### PR DESCRIPTION
**Pull request by maffoo**
Fixes #5418

### CODE REVIEW

**Comment by daxfohl** [comment]
I think the `repeat ` function is also affected.

File: cirq-core/cirq/circuits/circuit_operation.py [hunk 1]
```diff
@@ -148,14 +148,15 @@ def __post_init__(self):
...
-                object.__setattr__(self, 'repetition_ids', self._default_repetition_ids())
-            elif len(self.repetition_ids) != loop_size:
-                raise ValueError(
-                    f'Expected repetition_ids to be a list of length {loop_size}, '
-                    f'got: {self.repetition_ids}'
-                )
+            if self.use_repetition_ids:
+                loop_size = abs(self.repetitions)
+                if not self.repetition_ids:
+                    object.__setattr__(self, 'repetition_ids', self._default_repetition_ids())
```
Comment (daxfohl) [inline cirq-core/cirq/circuits/circuit_operation.py:154]: Could all these be consolidated by having `default_repetition_ids` return None if `use_repetition_ids ` is false?

Comment (maffoo) [reply cirq-core/cirq/circuits/circuit_operation.py]: Changed to do the `self.use_repetition_ids` check in `self._default_repetition_ids`.

**Comment by daxfohl** [comment]
lgtm

**Review by daxfohl — APPROVED** [review]

**Review by dstrain115 — APPROVED** [review]

**Pull request merged**